mimic/model/maas_objects.py

```diff
@@ -797,3 +849,13 @@ def latest_alarm_states_for_entity(self, entity_id):
         for state in alarm_states_for_entity:
             latest_alarm_states_by_alarm[state.alarm_id] = state
         return latest_alarm_states_by_alarm.values()
+
+    def list_connections_for_agent(self, agent_id):
+        """
+        Lists connections for an agent, or returns empty list if there is no such agent.
+        """
+        try:
+            agent = self.agents[self.agents.index(Matcher(lambda agent: agent.id == agent_id))]
```

Comment:
Looking at this line of code sparked a thought.  We have a general pattern throughout Mimic where we have data structures that involve lookups on one or more attributes.  There's almost always a "lookup by ID", but there's also often a "lookup by name" or "lookup by relationship to some other object".  It might be interesting to write a small general utility for this sort of structure; that would also help a lot if we wanted to start persisting things across mimic restarts because it would be a single point to go update.

Anyway, not really relevant to this particular PR, but worth thinking about for the future.